### PR TITLE
feat: support retry test api in BidiWriteObject

### PR DESCRIPTION
**Pull request by cojenco**
This adds retry test api support in gRPC BidiWriteObject

### CODE REVIEW

File: tests/test_testbench_retry.py [hunk 1]
```diff
@@ -786,6 +786,78 @@ def test_grpc_return_error_after_bytes(self):
         self.assertEqual(blob.bucket, "projects/_/buckets/bucket-name")
         self.assertEqual(blob.size, 2 * UPLOAD_QUANTUM)
 
+    def test_grpc_bidiwrite_return_error_after_bytes(self):
+        # Setup two after-bytes errors to test injecting failures in
```
Comment (tritone) [inline tests/test_testbench_retry.py:790]: Do we need a test for initial-response errors as well? Or are existing tests sufficient for this?

File: tests/test_testbench_retry.py [hunk 2]
```diff
@@ -786,6 +786,78 @@ def test_grpc_return_error_after_bytes(self):
...
+            ),
+            finish_write=True,
+        )
+        streamer = self.grpc.BidiWriteObject([r2], context)
+        responses = list(streamer)
+        context.abort.assert_called_with(StatusCode.UNAVAILABLE, unittest.mock.ANY)
+        blob = responses[0].resource
+        self.assertEqual(blob.name, "object-name")
+        self.assertEqual(blob.bucket, "projects/_/buckets/bucket-name")
+        self.assertEqual(blob.size, 2 * UPLOAD_QUANTUM)
```
Comment (tritone) [inline tests/test_testbench_retry.py:859]: Should this also include checking that the upload succeeds after the errors have been fully used up?

**Review by tritone — APPROVED** [review]
Looks good, couple minor questions re: testing.

File: tests/test_testbench_retry.py (hunk 2 above)
Comment (cojenco) [reply tests/test_testbench_retry.py:861]: Thanks for the review! Yep added a check that the upload succeeds with the accurate size

File: tests/test_testbench_retry.py (hunk 1 above)
Comment (cojenco) [reply tests/test_testbench_retry.py]: There are existing tests for initial-response errors. It doesn't hurt to add an extra check here, updated! Will be merging shortly

**Review by tritone — APPROVED** [review]

**Pull request merged**